src/docfx/build/xref/InternalXrefMapBuilder.cs

```diff
@@ -137,12 +143,29 @@ private InternalXrefSpec[] AggregateXrefSpecs(string uid, InternalXrefSpec[] spe
                 _errors.Add(Errors.Versioning.MonikerOverlapping(uid, specsWithSameUid.Select(spec => spec.DeclaringFile).ToList(), overlappingMonikers));
             }
 
-            return specsWithSameUid
+            var specs = specsWithSameUid
                    .OrderByDescending(spec => spec.Monikers.HasMonikers
                         ? spec.Monikers.Select(moniker => _monikerProvider.GetMonikerOrder(moniker)).Max()
-                        : int.MaxValue)
-                   .ThenBy(spec => spec.DeclaringFile)
-                   .ToArray();
+                        : int.MaxValue);
+
+            // TODO: clean up after new loc pipeline
+            specs = _config.IsLearn
+                ? specs.ThenByDescending(spec => GetUpdateTime(spec.DeclaringFile))
+                : specs.ThenBy(spec => spec.DeclaringFile);
```

Comment:
if two files are of the same latest update time, should we order by file name?

Reply:
Good thought! Though that would make output stable, but considering the loc deletion case, random picking one could better serve that purpose.